src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,416 @@
+      q = Int(sqrt(order(F)))
+      Z = identity_matrix(F,n)
+      # turn the elements on the main diagonal into 1
+      for i in 1:n
+         if A[i,i]!=0
+            lambda = _disc_log(w^(q+1),A[i,i])
+            Z[i,i] = w^-lambda
+            A[i,i] = 1
+         end
+      end
+      D = Z*D
+      # moving all hyperbolic lines at the end
+      Z = identity_matrix(F,n)
+      our_permut = Array(1:n)
+      NOZ = 0      # Number Of Zeros on the diagonal before A[i,i]
+      for i in 1:n
+         if A[i,i]==0
+            NOZ += 1
+         else
+            j = i
+            while j>i-NOZ
+               swap_cols!(A,j,j-1)
+               j -= 1
+            end
+            j = i
+            while j > i-NOZ
+               swap_rows!(A,j,j-1)
+               j -= 1
+            end
+            for j in 1:NOZ
+               our_permut[i+1-j] = our_permut[i-j]
+            end
+            our_permut[i-NOZ] = i
+         end
+      end
+      Z = permutation_matrix(F, our_permut)
+      D = Z*D
+      # turn 2x2 identities into 2x2 anti-diagonal blocks
+      Z = identity_matrix(F,n)
+      if isodd(q)
+         b = (1+w^(div((q-1)^2,2)))^-1
+         a = b*w^(div((1-q),2))
+         d = 1
+         c = w^(div((q-1),2))
+      else
+         b = (1+w^(q-1))^-1
+         a = b*w^(q-1)
+         d = 1
+         c = 1
+      end
+      S = matrix(F,2,2,[a,b,c,d])
+      if div(n-NOZ,2)==0
+         S = zero_matrix(F,0,0)
+      else
+         S = diagonal_join([S for i in 1:div(n-NOZ,2)])
+      end
+      # turn into standard GAP form
+      sec_perm = Int[]
+      for i in 1:div(n,2)
+         sec_perm = vcat([i,n+1-i],sec_perm)
+      end
+      if isodd(n)
+         insert_block!(Z,S,2,2)
+         sec_perm = vcat([div(n+1,2)],sec_perm)
+      else
+         insert_block!(Z,S,1,1)
+      end
+      D = transpose(permutation_matrix(F,sec_perm))*Z*D
+   end
+
+   return D
+end
+
+
+
+###############################################################################################################
+
+# Change of basis between two matrices
+
+###############################################################################################################
+
+# modifies A by eliminating all hyperbolic lines and turning A into a diagonal matrix
+# return the matrix Z such that Z*A*transpose(Z) is diagonal
+function _elim_hyp_lines(A::MatElem{T}) where T <: FinFieldElem
+   F = base_ring(A)
+   n = nrows(A)
+   b = matrix(F,2,2,[1,1,1,-1])  # change of basis from matrix([0,1,1,0]) to matrix([2,0,0,-2])
+   Z = identity_matrix(F,n)
+
+   i = 1
+   while i <= n
+      if A[i,i]==0
+         A[i,i]=2
+         A[i,i+1]=0
+         A[i+1,i]=0
+         A[i+1,i+1]=-2
+         insert_block!(Z,b,i,i)
+         i+=2
+      else
+         i+=1
+      end
+   end
+
+   return Z
+end
+
+
+# return true, D such that D*B2*conjugatetranspose(D)=B1
+# return false, nothing if D does not exist
+# TODO: orthogonal only in odd char, at the moment
```

Comment:
Then please add a check in the code that throws an `error("even char not yet supported")` or so when the characteristic is even